src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -915,16 +916,23 @@ private void GenerateIndexerSetter(Function function)
             }
             else
             {
-                WriteLine("*({0}.Internal*) Internal.{1}({2}, {3}) = *({0}.Internal*) value.{2};",
-                    type.ToString(), internalFunction,
-                    Helpers.InstanceIdentifier, function.Parameters[0].Name);
+                Class @class;
+                if(type.TryGetClass(out @class) && @class.IsValueType)
+                    WriteLine("*({0}.Internal*) Internal.{1}({2}, {3}) = ({0}.Internal) value.{2};",
+                        type.ToString(), internalFunction,
+                        Helpers.InstanceIdentifier, function.Parameters[0].Name);
+                else
+                    WriteLine("*({0}.Internal*) Internal.{1}({2}, {3}) = *({0}.Internal*) value.{2};",
+                        type.ToString(), internalFunction,
+                        Helpers.InstanceIdentifier, function.Parameters[0].Name);
```

Comment:
@ddobrev  @tritao Should I use braces here or is it fine like for some of the other one liners?